livebook_examples/rtmp/rtmp_sender.livemd

```diff
@@ -0,0 +1,140 @@
+
+```elixir
+defmodule RTMP.Sender.Pipeline do
+  use Membrane.Pipeline
+
+  @samples_url "https://raw.githubusercontent.com/membraneframework/static/gh-pages/samples/big-buck-bunny/"
+  @video_url @samples_url <> "bun33s_480x270.h264"
+  @audio_url @samples_url <> "bun33s.aac"
+  @impl true
+  def handle_init(_ctx, destination: destination) do
+    video_source =
+      child(:video_source, %Membrane.Hackney.Source{
+        location: @video_url,
+        hackney_opts: [follow_redirect: true]
+      })
+      |> child(:video_parser, %Membrane.H264.FFmpeg.Parser{
+        framerate: {25, 1},
+        alignment: :au,
+        attach_nalus?: true,
+        skip_until_keyframe?: true
+      })
+      |> child(:video_realtimer, Membrane.Realtimer)
+      |> child(:video_payloader, Membrane.MP4.Payloader.H264)
+
+    audio_source =
+      child(:audio_source, %Membrane.Hackney.Source{
+        location: @audio_url,
+        hackney_opts: [follow_redirect: true]
+      })
+      |> child(:audio_parser, %Membrane.AAC.Parser{
+        in_encapsulation: :ADTS,
+        out_encapsulation: :ADTS
+      })
+      |> child(:audio_realtimer, Membrane.Realtimer)
+
+    rtmp_sink =
+      child(:rtmp_sink, %Membrane.RTMP.Sink{
+        rtmp_url: destination,
+        max_attempts: :infinity
+      })
+
+    structure = [
+      video_source
+      |> via_in(:video)
+      |> get_child(:rtmp_sink),
+      audio_source
+      |> via_in(:audio)
+      |> get_child(:rtmp_sink),
+      rtmp_sink
+    ]
+
+    {[spec: structure, playback: :playing], %{streams_to_end: 2}}
+  end
+
+  # The rest of the example module is only used for self-termination of the pipeline after processing finishes
+
+  @impl true
+  def handle_element_end_of_stream(:rtmp_sink, _pad, _ctx, %{streams_to_end: 1} = state) do
+    Membrane.Pipeline.terminate(self())
+    {[], %{state | streams_to_end: 0}}
+  end
+
+  @impl true
+  def handle_element_end_of_stream(:rtmp_sink, _pad, _ctx, state) do
+    {[], %{state | streams_to_end: 1}}
+  end
+
+  @impl true
+  def handle_element_end_of_stream(_child, _pad, _ctx, state) do
+    {[], state}
+  end
+end
+
+:ok
+```
+
+## Sender
+
+RTMP protocol requires a client-server communication, where the TCP server receives the data and the client sends it.
+
+```elixir
+defmodule RTMP.Sender do
+  def run(port: port) do
+    destination_url = "rtmp://localhost:#{port}"
+
+    {:ok, pipeline} = start_server(destination_url)
+
+    await_termination(pipeline)
+  end
+
+  defp start_server(destination_url) do
```

Comment:
Since we're no longer calling this a server in the linie above the code, I would rename this, perhaps to `start_tcp_client`?